src/main/groovy/io/codearte/gradle/nexus/infra/SimplifiedHttpJsonRestClient.groovy

```diff
@@ -20,58 +23,59 @@ class SimplifiedHttpJsonRestClient {
+        clientBuilder.addInterceptor(new Interceptor() {
+            @Override
+            Response intercept(@NotNull Interceptor.Chain chain) throws IOException {
+                Request.Builder request = chain.request().newBuilder()
+                    .header("Content-Type", "application/json")
+                    .header("Accept", "application/json")
+
+                if (username != null && password != null) {
+                    request.header("Authorization", Credentials.basic(username, password))
+                }
+
+                return chain.proceed(request.build())
+            }
+        })
+        this.restClient = clientBuilder.build()
     }
 
     Map get(String uri) {
-        return (Map) sendRequestHandlingErrors(uri, null, restClient.&get, RequestType.GET).data
+        Request request = new Request.Builder()
+            .url(uri)
+            .build()
+        return sendRequestHandlingErrors(request, RequestType.GET)
     }
 
     Map post(String uri, Map content) {
-        return (Map) sendRequestHandlingErrors(uri, content, restClient.&post, RequestType.POST).data
+        Request request = new Request.Builder()
+            .method("POST", RequestBody.create(JsonOutput.toJson(content), null))
+            .url(uri)
+            .build()
+        return sendRequestHandlingErrors(request, RequestType.POST)
     }
 
-    private HttpResponseDecorator sendRequestHandlingErrors(String uri, Map content, Closure<Object> clientMethodHandler, RequestType requestTypeName) {
+    private Map sendRequestHandlingErrors(Request request, RequestType requestTypeName) {
         try {
-            return prepareAndSendRequest(uri, content, clientMethodHandler, requestTypeName)
-        } catch (HttpResponseException e) {
-            HttpResponseDecorator resp = e.getResponse();
-            String message = "${resp.statusLine.statusCode}: ${resp.statusLine.reasonPhrase}, body: ${resp.data ?: '<empty>'}"
-            //TODO: Suppress error message on 404 if waiting for drop?
-            log.warn("$requestTypeName request failed. ${message}")
-            throw new NexusHttpResponseException(e.getStatusCode(), message, e)
-        }
-    }
-
-    private HttpResponseDecorator prepareAndSendRequest(String uri, Map content, Closure<Object> clientMethodHandler, RequestType requestType) {
-        setUriAndAuthentication(uri)
-        Map params = createAndInitializeCallParametersMap()
-        if (content != null) {
-            params.body = content
-            log.debug("$requestType request content: $content")
+            return restClient.newCall(request).execute().withCloseable {
+                if (!it.successful) {
+                    String message = "${it.code()}: ${it.message()}, body: ${it.body().string() ?: '<empty>'}"
+                    //TODO: Suppress error message on 404 if waiting for drop?
+                    log.warn("$requestTypeName request failed. ${message}")
```

Comment:
`requestTypeName` seems to be used only here. Could you check if `request.method()` couldn't be used instead?

Reply:
Nice!